python/cudf/cudf/tests/test_parquet.py

```diff
@@ -1140,6 +1140,74 @@ def test_parquet_reader_struct_basic(tmpdir, data):
     assert expect.equals(got.to_arrow())
 
 
+def select_columns_params():
+    dfs = [
+        # struct
+        (
+            [
+                {"a": 1, "b": 2},
+                {"a": 10, "b": 20},
+                {"a": None, "b": 22},
+                {"a": None, "b": None},
+                {"a": 15, "b": None},
+            ],
+            [["struct"], ["struct.a"], ["struct.b"], ["c"]],
+        ),
+        # struct-of-list
+        (
+            [
+                {"a": 1, "b": 2, "c": [1, 2, 3]},
+                {"a": 10, "b": 20, "c": [4, 5]},
+                {"a": None, "b": 22, "c": [6]},
+                {"a": None, "b": None, "c": None},
+                {"a": 15, "b": None, "c": [-1, -2]},
+                None,
+                {"a": 100, "b": 200, "c": [-10, None, -20]},
+            ],
+            [
+                ["struct"],
+                ["struct.c"],
+                ["struct.c.list"],
+                ["struct.c.list.item"],
+                ["struct.b", "struct.c"],
+                ["struct.b", "struct.d", "struct.c"],
+            ],
+        ),
+        # list-of-struct
+        (
+            [
+                [{"a": 1, "b": 2}, {"a": 2, "b": 3}, {"a": 4, "b": 5}],
+                None,
+                [{"a": 10, "b": 20}],
+                [{"a": 100, "b": 200}, {"a": None, "b": 300}, None],
+            ],
+            [
+                ["struct"],
+                ["struct.list"],
+                ["struct.list.item"],
+                ["struct.list.item.a", "struct.list.item.b"],
+                ["struct.list.item.c"],
+            ],
+        ),
+    ]
+    for df_col_pair in dfs:
+        for cols in df_col_pair[1]:
+            yield df_col_pair[0], cols
+
+
+@pytest.mark.parametrize("data, columns", select_columns_params())
+def test_parquet_reader_struct_select_columns1(tmpdir, data, columns):
+    table = pa.Table.from_pydict({"struct": data})
+    fname = tmpdir.join("test_parquet_reader_struct_basic.parquet")
+
+    pa.parquet.write_table(table, fname)
+    assert os.path.exists(fname)
+
+    expect = pq.ParquetFile(fname).read(columns=columns)
+    got = cudf.read_parquet(fname, columns=columns)
+    assert expect.equals(got.to_arrow())
```

Comment:
```suggestion
    buff = BytesIO()

    pa.parquet.write_table(table, buff)
    assert os.path.exists(buff)

    expect = pq.ParquetFile(buff).read(columns=columns)
    got = cudf.read_parquet(buff, columns=columns)
    assert expect.equals(got.to_arrow())
```

Reply:
Just curious. Is BytesIO to be preferred for all such tests from now?

Reply:
Unless you're specifically testing file IO, buffer input is preferred. The only reason I;m aware of is that the tests should be faster without file IO.

Reply:
Are you sure we also need the line `assert os.path.exists(buff)`?

Reply:
that seems like a typo

Reply:
Yeah, typo. Ignore `assert os.path.exists(buff)`